modules/standard/IO.chpl

```diff
@@ -2134,50 +2115,158 @@ record channel {
   var _readWriteThisFromLocale = nilLocale;
 }
 
+/*
+
+A ``fileReader`` supports sequential reading from an underlying :record:`file`
+object. It can buffer data. Read operations on it might return old data. Use
+:proc:`channel.flush` to control this buffering.
+
+The :record:`fileReader` type is implementation-defined.
+A value of the :record:`fileReader` type refers to the state that is used
+to implement the reading operations.
+
+When a :record:`fileReader` formal argument has default intent, the actual is
+passed by ``const ref`` to the formal upon a function call, and the formal
+cannot be assigned within the function.
+
+The default value of the :record:`fileReader` type is not associated
+with any file, and so cannot be used to perform I/O.
+
+The :record:`fileReader` type is generic.
+
+The :record:`fileReader` type supports 3 fields:
+
+ * ``param writing: bool = false``:
+
+ *  ``param kind:iokind``:
+     kind is an enum :type:`iokind` that allows narrowing
+     this channel's I/O style for more efficient binary I/O.
+
+  * ``param locking:bool``:
+     locking is a boolean indicating whether it is safe to use this
+     channel concurrently (when `true`).
+*/
+type fileReader = _channel(writing=false, ?);
```

Comment:
In looking through the documentation, I noticed that this is one of the few places `_channel` turns up.  Would you be up for using split init to hide it?